src/pineko/check.py

```diff
@@ -106,58 +140,25 @@ def is_fonll_b(fns, lumi):
     return False
 
 
-def contains_fact(grid, max_as, max_al):
-    """Check whether fact scale-variations are available.
-
-    The order is specified by max_as and max_al.
-
-    Parameters
-    ----------
-    grid : pineappl.grid.Grid
-           Pineappl grid
-    max_as : int
-             max as order
-    max_al : int
-             max al order
-    Returns
-    -------
-    bool
-        is fact scale-variation available for as
-    bool
-        is fact scale-variation available for al
-    """
+def get_order_list(grid, max_as, max_al):
+    """Return the order list after filtering according to max_as and max_al."""
     order_array = np.array([order.as_tuple() for order in grid.orders()])
-    order_mask = pineappl.grid.Order.create_mask(grid.orders(), max_as, max_al)
+    order_mask = pineappl.grid.Order.create_mask(grid.orders(), max_as, max_al, True)
     order_list = order_array[order_mask]
+    return order_list
+
+
+def pure_qcd_orders(order_list):
+    """Return the pure alpha_s orders appearing in the grid."""
     as_orders = []
-    al_orders = []
+    min_al = min(ord[1] for ord in order_list)
     for order in order_list:
-        as_orders.append(order[0])
-        al_orders.append(order[1])
-    min_as = min(as_orders)
-    min_al = min(al_orders)
-    order_as_is_present = False
-    order_al_is_present = False
-    sv_as_present = False
-    sv_al_present = False
-    for order in order_list:
-        # fact sv starts at NLO with respect to the first non zero order
-        if order[0] == min_as + 1:
-            order_as_is_present = True
-            if order[-1] != 0:
-                sv_as_present = True
-        if order[1] == min_al + 1:
-            order_al_is_present = True
-            if order[-1] != 0:
-                sv_al_present = True
-    if not order_as_is_present:
-        sv_as_present = True
-    if not order_al_is_present:
-        sv_al_present = True
-    return sv_as_present, sv_al_present
-
-
-def contains_ren(grid, max_as, max_al):
+        if order[1] == min_al:
+            as_orders.append(order)
+    return as_orders
```

Comment:
For this a list comprehension might be appropriate:
```py
    min_al = min(ord[1] for ord in order_list)
    return [ord for ord in orders if ord[1] == min_al]
```

I know it is just syntactic sugar, but it seems extremely clear, so I'd exploit it :)

Reply:
[4a1066b](https://github.com/NNPDF/pineko/pull/75/commits/4a1066bcdbc04b4f3deb420914620b230dc9c38b)